packages/next/src/utils.ts

```diff
@@ -2731,6 +2760,37 @@ export async function getMiddlewareBundle({
   };
 }
 
+/**
+ * Attempts to read the functions config manifest from the pre-defined
+ * location. If the manifest can't be found it will resolve to
+ * undefined.
+ */
+export async function getFunctionsConfigManifest(
+  entryPath: string,
+  outputDirectory: string
+): Promise<FunctionsConfigManifest | undefined> {
+  const functionConfigManifestPath = path.join(
+    entryPath,
+    outputDirectory,
+    './server/functions-config-manifest.json'
+  );
+
+  const hasManifest = await fs
+    .access(functionConfigManifestPath)
+    .then(() => true)
+    .catch(() => false);
+
+  if (!hasManifest) {
+    return;
+  }
+
+  const manifest = (await fs.readJSON(
+    functionConfigManifestPath
+  )) as FunctionsConfigManifest;
+
+  return manifest;
```

Comment:
Any concern for lack of JSON validation here?

Reply:
We could validate for the right types, though I wouldn't do strict validation as upstream changes in Next.js may break builds then.

Reply:
@ecklf The manifest file should have a `version` prop

Reply:
@styfle I've asked @feedthejim about this too, but not all manifest files use a `version` prop. Is this absolutely required (we don't expect the signature of this to change).

Reply:
Next.js PR here https://github.com/vercel/next.js/pull/52507